src/iceberg/catalog/rest/CMakeLists.txt

```
@@ -0,0 +1,54 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+if(NOT ICEBERG_BUILD_REST_CATALOG)
+  return()
+endif()
+
+set(ICEBERG_REST_CATALOG_SOURCES rest_catalog.cc)
+
+set(ICEBERG_REST_CATALOG_STATIC_BUILD_INTERFACE_LIBS)
+set(ICEBERG_REST_CATALOG_SHARED_BUILD_INTERFACE_LIBS)
+set(ICEBERG_REST_CATALOG_STATIC_INSTALL_INTERFACE_LIBS)
+set(ICEBERG_REST_CATALOG_SHARED_INSTALL_INTERFACE_LIBS)
+
+list(APPEND ICEBERG_REST_CATALOG_STATIC_BUILD_INTERFACE_LIBS
+     "$<IF:$<TARGET_EXISTS:iceberg_static>,iceberg_static,iceberg_shared>" cpr::cpr)
+list(APPEND ICEBERG_REST_CATALOG_SHARED_BUILD_INTERFACE_LIBS
+     "$<IF:$<TARGET_EXISTS:iceberg_shared>,iceberg_shared,iceberg_static>" cpr::cpr)
+list(APPEND
+     ICEBERG_REST_CATALOG_STATIC_INSTALL_INTERFACE_LIBS
+     "$<IF:$<TARGET_EXISTS:Iceberg::iceberg_static>,Iceberg::iceberg_static,Iceberg::iceberg_shared>"
+     "$<IF:$<BOOL:${CPR_VENDORED}>,Iceberg::cpr,cpr::cpr>")
+list(APPEND
+     ICEBERG_REST_CATALOG_SHARED_INSTALL_INTERFACE_LIBS
+     "$<IF:$<TARGET_EXISTS:Iceberg::iceberg_shared>,Iceberg::iceberg_shared,Iceberg::iceberg_static>"
+     "$<IF:$<BOOL:${CPR_VENDORED}>,Iceberg::cpr,cpr::cpr>")
+
+add_iceberg_lib(iceberg_rest_catalog
```

```suggestion
add_iceberg_lib(iceberg_rest
```

Let's make it shorter